app.rb

```diff
@@ -13,20 +15,23 @@ class CondaAPI < Sinatra::Base
     Conda.instance.package_names.to_json
   end
 
-  get "/packages/:name" do
-    content_type :json
-    package_version("pkgs/main", params[:name])
-  end
-
-  get "/packages/:channel/:name" do
-    channel = CGI.unescape(CGI.unescape(params[:channel]))  # %252F -> %2F -> /
+  get "/package" do
     content_type :json
-    package_version(channel, params[:name])
+    if request.query_string.empty?
+      {"error" => "Please provide at least a package name ?name= parameter"}.to_json
+    else
+      package_version(request.query_string)
+    end
   end
 
   private
 
-  def package_version(channel, name)
+  def package_version(query_string)
+    # Parse the query string for channel and name
+    qs = CGI::parse(query_string)
```

Comment:
Sinatra handles query params the same as URL params. No need to do this parsing.

Reply:
It doesn't seem to. ![](https://tyrelsouza.com/screenshots/29d633b22d3ee9ec11bbb3612d310b97.png)

Reply:
Ignore, this was the wrong test, phew.